react/src/Root/App/Information/GeohazardAffectedHouseholds/DonutChart/index.tsx

```diff
@@ -0,0 +1,163 @@
+import React from 'react';
+import { _cs } from '@togglecorp/fujs';
+import memoize from 'memoize-one';
+
+import List from '#rscv/List';
+import connectWithStyles from '#rsu/styles/connectWithStyles';
+import Responsive from '#rscg/Responsive';
+
+import Arc from './Arc';
+import styles from './styles.scss';
+
+interface SelectorFunction<T, V = string> {
+    (value: T): V;
+}
+
+interface Props<T> {
+    className?: string;
+    data?: T[];
+    boundingClientRect: {
+        width?: number;
+        height?: number;
+    };
+    keySelector: SelectorFunction<T, string | number>;
+    valueSelector: SelectorFunction<T, number>;
+    labelSelector: SelectorFunction<T, string | number>;
+    colorSelector: SelectorFunction<T, string>;
+    currentStyles: object;
+}
+
+interface RenderDatum {
+    key: string | number;
+    value: number;
+    label: string | number;
+    color: string;
+    startAngle: number;
+    endAngle: number;
+}
+
+const arcRenderKeySelector = (datum: RenderDatum) => (
+    datum.key as string
```

Comment:
Why cast to string?

Reply:
Because key selector does not accept number

Reply:
Will be fixed in https://github.com/toggle-corp/react-store/pull/599